### DISCUSSION TITLE
Beta Launch (ICON-BSC): Smoke Test Checklist

### DISCUSSION THREAD

**Discussion by bbist** [icon-project/icon-bridge#314] [Ideas]
This discussion will include checklist for smoke test cases on the primary icon-bridge functionalities. Please make suggestions on comments.

Following flow tests should include:
1. Successful transfer cases
2. Failed transfer + Refund cases
3. Verification of fee gathering on fee aggregator wallet address
4. Sending tokens (native, wrapped, external) to contracts: 
    1. External (Tokens)
    2. BTP (BTS, BMC)

- [x] Native Token Transfers
  - [x] ICX: ICON -> BSC
  - [x] BNB: BSC -> ICON
  
- [x] Non-Native Token Transfers
  - [x] sICX: ICON -> BSC
  - [x] bnUSD:  ICON -> BSC
  - [x] BUSD: BSC -> ICON
  - [x] USDT: BSC -> ICON
  - [x] USDC: BSC -> ICON
  - [x] BTCB: BSC -> ICON
  - [x] ETH: BSC -> ICON

- [x] Wrapped Token Transfers
  - [x] ICX: BSC -> ICON
  - [x] sICX: BSC -> ICON
  - [x] bnUSD:  BSC -> ICON
  - [x] BNB: ICON -> BSC
  - [x] BUSD: ICON -> BSC 
  - [x] USDT: ICON -> BSC
  - [x] USDC: ICON -> BSC
  - [x] BTCB: ICON -> BSC
  - [x] ETH: ICON -> BSC

- [ ] Blacklist Checks
  - [ ] Blacklist ICON Address
  - [ ] Blacklist BSC Address
  - [ ] Transfer token from BSC to ICON: ICON Address blacklisted
  - [ ] Transfer token from ICON to BSC: ICON Address blacklisted
  - [ ] Transfer token from BSC to ICON: BSC Address blacklisted
  - [ ] Transfer token from ICON to BSC: BSC Address blacklisted (Should fail on ICON side itself)
  - [ ] Try to blacklist invalid address on BSC
 
- [ ] Token Limit Checks
  - [ ] Set token limit for registered tokens
  - [ ] Set token limit for not registered tokens
  - [ ] Check if token limit is consistent across BSC and ICON
  - [ ] Transfer from ICON to BSC which exceeds token limit

- [ ] Fee Verification
  - [ ] Verify if fee = amount * feeNumerator / 1000 + fixedFee during token transfer

- [ ] Other cases
  - [ ] External user tries to call `handleBTPMessage `
  - [ ] External user tries to call `handleRelayMessage`

**Comment by CyrusVorwald** [comment]
I think the tests should also include what happens when the relay goes down in either direction during various steps in a transfer. This may already be covered by what you describe as fail transfer cases.

It should also include what the possible outcomes are when an external user calls any public methods in the smart contracts. You mentioned handleBTPMessage and handleRelayMessage. Are there any others? Are there multiple possible inputs and control flows in each?
